In [1]:
import pandas as pd
import numpy as np

### **Dataset Overview**
- **Total Entries**: 100,000 transactions
- **Columns**:
  - `TransactionID` (int): Unique identifier for each transaction
  - `TransactionDate` (object): Timestamp of the transaction
  - `Amount` (float): Transaction amount
  - `MerchantID` (int): ID of the merchant
  - `TransactionType` (object): Type of transaction (e.g., purchase, refund)
  - `Location` (object): Location where the transaction occurred
  - `IsFraud` (int): **Target variable** (0 = legitimate, 1 = fraud)

let's proceed! 🚀

In [ ]:
# Load data from a CSV file
data = pd.read_csv('credit_card_fraud_dataset.csv')


#### Check for Missing Values & Data Distribution

In [5]:
# Display basic info and first few rows
df_info = data.info()
df_head = data.head()

df_info, df_head, data.describe()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 7 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   TransactionID    100000 non-null  int64  
 1   TransactionDate  100000 non-null  object 
 2   Amount           100000 non-null  float64
 3   MerchantID       100000 non-null  int64  
 4   TransactionType  100000 non-null  object 
 5   Location         100000 non-null  object 
 6   IsFraud          100000 non-null  int64  
dtypes: float64(1), int64(3), object(3)
memory usage: 5.3+ MB


(None,
    TransactionID             TransactionDate   Amount  MerchantID  \
 0              1  2024-04-03 14:15:35.462794  4189.27         688   
 1              2  2024-03-19 13:20:35.462824  2659.71         109   
 2              3  2024-01-08 10:08:35.462834   784.00         394   
 3              4  2024-04-13 23:50:35.462850  3514.40         944   
 4              5  2024-07-12 18:51:35.462858   369.07         475   
 
   TransactionType      Location  IsFraud  
 0          refund   San Antonio        0  
 1          refund        Dallas        0  
 2        purchase      New York        0  
 3        purchase  Philadelphia        0  
 4        purchase       Phoenix        0  ,
        TransactionID         Amount     MerchantID        IsFraud
 count  100000.000000  100000.000000  100000.000000  100000.000000
 mean    50000.500000    2497.092666     501.676070       0.010000
 std     28867.657797    1442.415999     288.715868       0.099499
 min         1.000000       1.050000  

**Missing Values:**  No missing values in any column.

**Numerical Data Summary:**

Transaction Amount:

`Mean`: $2,497.09

`Min`: $1.05, Max: $4,999.77
- Distribution seems fairly spread out.

`IsFraud`:
- Only 1% of transactions are fraud (high class imbalance).

**Categorical Data Summary:**

TransactionType: Only 2 categories (purchase, refund).

Location: 10 unique locations, with Chicago being the most common.